src/redux/reducers/resereveSlice.js

```diff
@@ -0,0 +1,65 @@
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
+import axios from 'axios';
+
+const baseUrl = process.env.REACT_APP_API_AUTH_URL;
+const endpoint = '/api/v1/reservations/';
+const url = baseUrl + endpoint;
+const token = JSON.parse(localStorage.getItem('token'));
+
+const headers = {
+  'Content-Type': 'application/json',
+  Authorization: `Bearer ${token}`,
+};
+
+const initialState = {
+  cities: null,
+  isCreated: false,
+};
+
+const createReservation = createAsyncThunk(
+  'reservation/createReservation',
+  async (reservationData, { rejectWithValue }) => {
+    try {
+      const response = await axios.post(url, reservationData, { headers });
+      return response.data;
+    } catch (error) {
+      // Log the error for debugging purposes
+      console.error('Error:', error);
```

Comment:
please rewrite.

Reply:
@lincoln1883 OK, let me check and rewrite to handle the specific errors to show to the users.